protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java

```diff
@@ -36,6 +38,77 @@
 public class HintRetrieveRequestTest {
 
     @Test
+    public void testFromAuthMethods() {
+        HintRetrieveRequest request = HintRetrieveRequest.fromAuthMethods(
+                AuthenticationMethods.EMAIL,
+                AuthenticationMethods.FACEBOOK);
+
+        assertThat(request.getAuthenticationMethods().size()).isEqualTo(2);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.EMAIL);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.FACEBOOK);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testFromAuthMethods_noneProvided() {
+        HintRetrieveRequest.fromAuthMethods();
+    }
+
+    @Test
+    public void testFromAuthMethods_withSet() {
+        HintRetrieveRequest request = HintRetrieveRequest.fromAuthMethods(
+                ImmutableSet.of(AuthenticationMethods.EMAIL, AuthenticationMethods.FACEBOOK));
+
+        assertThat(request.getAuthenticationMethods().size()).isEqualTo(2);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.EMAIL);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.FACEBOOK);
+    }
+
+    @Test(expected=IllegalArgumentException.class)
+    public void testFromAuthMethods_emptySet() {
+        HintRetrieveRequest request = HintRetrieveRequest.fromAuthMethods(
+                Collections.emptySet());
+
```

Comment:
Test should have failed before these lines are run. Remove please.

Reply:
I do love to copy paste. Done.